.github/workflows/CI.yml

```diff
@@ -14,41 +14,79 @@
 # limitations under the License.
 
 ---
-name: Code Formatting
+name: Daffodil CI
 on:
   push:
     branches-ignore: [ 'dependabot/**' ]
   pull_request:
     types: [opened, synchronize, reopened]
 
 jobs:
-  ts-format:
-    name: TypeScript Formatting
+  single-commit:
+    name: Single Commit Pull Request
+    if: github.event_name == 'pull_request'
     runs-on: ubuntu-20.04
     steps:
-      - uses: actions/checkout@v2.3.5
-      - name: Setup Node
-        uses: actions/setup-node@v1.4.6
+      - name: Check Single Commit
+        uses: actions/github-script@v5.0.0
         with:
-          node-version: '16'
-      - run: yarn install
-      - run: yarn lint
+          script: |
+            const commits = await github.rest.pulls.listCommits({
+              ...context.repo,
+              pull_number: context.issue.number,
+            });
+            core.info("Number of commits in this pull request: " + commits.data.length);
+            if (commits.data.length > 1) {
+              core.setFailed("If approved with two +1's, squash this pull request into one commit");
+            }
 
-  scala-format:
-    name: Scala Formatting
+  format-rat-check:
+    name: TypeScript/Scala Formatting and Rat Check
     runs-on: ubuntu-20.04
     defaults:
       run:
         shell: bash
     env:
       SBT: sbt -J-Xms1024m -J-Xmx5120m -J-XX:ReservedCodeCacheSize=512m -J-XX:MaxMetaspaceSize=1024m ++2.12.13
-    
+
     steps:
       - uses: actions/checkout@v2.3.5
+      - name: Setup Node
+        uses: actions/setup-node@v1.4.6
+        with:
+          node-version: '16'
       - name: Setup Java
         uses: actions/setup-java@v2.3.1
         with:
           distribution: temurin
           java-version: 11
       - run: $SBT scalafmtCheck
       - run: $SBT scalafmtSbtCheck
+      - run: yarn install
+      - run: yarn lint
+      - name: Run Rat Check
+        run: $SBT ratCheck || (cat target/rat.txt; exit 1)
+      
+  build-test-package:
+    name: Build, Test, and Package
+    strategy:
+      matrix:
+        os: [macos-11, ubuntu-20.04, windows-2019]
+        node: [ '10', '12', '14', '16' ]
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash
+    env:
+      SBT: sbt -J-Xms1024m -J-Xmx5120m -J-XX:ReservedCodeCacheSize=512m -J-XX:MaxMetaspaceSize=1024m ++2.12.13
+    steps:
+      - uses: actions/checkout@v2.3.5
+      - name: Install Node.js
+        uses: actions/setup-node@v1.4.6
+        with:
+          node-version: ${{ matrix.node}}
+      - run: yarn build
+      - run: yarn test
+      - run: $SBT test
+      - run: yarn package
+      - run: $SBT package
```

Comment:
Looks like `yarn package` is new to the CI and it runs `vsce package`. Is that any different than `yarn build`?

Also, `yarn build` will run `sbt universal:packageBin`, which will runs `sbt package`, so I think the sbt package line is redundant.